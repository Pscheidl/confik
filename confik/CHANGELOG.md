# Changelog

## Unreleased

- Attributes that receive expressions (`default` and `from`) now need to be unquoted. E.g.:

  ```diff
  - struct MyStruct(#[config(default = "\"Hello World\"") String)
  + struct MyStruct(#[config(default = "Hello World") String)

  - struct MyStruct(#[config(default = "5usize") usize)
  + struct MyStruct(#[config(default = 5usize) usize)
  ```

## 0.7.0

- Initial release.
