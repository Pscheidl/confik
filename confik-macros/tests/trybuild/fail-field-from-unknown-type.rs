//! Check that the `from` attribute works
use confik::{Configuration, TomlSource};

#[derive(Debug, Configuration, PartialEq, Eq)]
struct Config {
    #[confik(from = "A")]
    param: String,
}

fn main() {
    let config = Config::builder()
        .override_with(TomlSource::new("param = 5"))
        .try_build()
        .expect("Failed to build with no required data");
    assert_eq!(
        Config {
            param: String::from("Hello world")
        },
        config
    );
}
