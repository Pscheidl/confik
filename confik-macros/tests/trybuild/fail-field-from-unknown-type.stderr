error[E0412]: cannot find type `A` in this scope
 --> tests/trybuild/fail-field-from-unknown-type.rs:6:21
  |
5 | struct Config {
  |              - help: you might be missing a type parameter: `<A>`
6 |     #[confik(from = A)]
  |                     ^ not found in this scope

error[E0412]: cannot find type `A` in this scope
 --> tests/trybuild/fail-field-from-unknown-type.rs:6:21
  |
6 |     #[confik(from = A)]
  |                     ^ not found in this scope

error[E0283]: type annotations needed
 --> tests/trybuild/fail-field-from-unknown-type.rs:6:21
  |
5 |   struct Config {
  |          ------ in this derive macro expansion
6 |       #[confik(from = A)]
  |  _____________________^
7 | |     param: String,
  | |_________^ cannot infer type
  |
  = note: cannot satisfy `_: Default`
  = note: this error originates in the derive macro `::std::default::Default` (in Nightly builds, run with -Z macro-backtrace for more info)
